index.js

```diff
@@ -1,17 +1,15 @@
+const getObject = (arr, obj) => obj ? {
+    id: obj.id,
+    next: obj.next,
+    value: obj.value || getObject(arr, arr.find(e => e.id === obj.next)),
+} : null;
+
+const arrayProblemSolver = arr => arr.map(obj => getObject(arr, obj));
+
 const array = [
-    {
-        id: 'one',
-        next: 'two',
-        value: null
-    },
-    {
-        id: 'two',
-        next: 'three',
-        value: null
-    },
-    {
-        id: 'three',
-        next: null,
-        value: null
-    },
-];
\ No newline at end of file
+    { id: 'one', next: 'two', value: null },
+    { id: 'two', next: 'three', value: null },
+    { id: 'three', next: null, value: null },
+];
+
+console.log(arrayProblemSolver(array));
```

Comment:
we can use **_JSON.stringify_** to convert the value into JSON string and that's the way, we can see the value of nested Objects on the terminal rather than just seeing it like this 

> value: { id: 'two', next: 'three', value: [Object] }